docs/_ruleset_functions.md

```diff
@@ -1,33 +1,270 @@
+
+      - Else, zone is unconditioned: ```else:```  
+
+        - For each surface in zone:  ```for surface in zone.surfaces:```  
+
+          - If surface adjacency is interior and the adjacent zone is conditioned or semi-heated, surface is classified as semi-exterior: ```if ( surface.adjacent_to == "INTERIOR" ) AND ( ( zone_conditioning_category_dict[surface.adjacent_zone_id] == "CONDITIONED RESIDENTIAL" ) OR ( zone_conditioning_category_dict[surface.adjacent_zone_id] == "CONDITIONED NON-RESIDENTIAL" ) OR ( zone_conditioning_category_dict[surface.adjacent_zone_id] == "CONDITIONED MIXED" ) OR ( zone_conditioning_category_dict[surface.adjacent_zone_id] == "SEMI-HEATED" ) ): surface_conditioning_category_dict[surface.id] = "SEMI-EXTERIOR"```  
+
+          - Else, surface is classified as unregulated: ```else: surface_conditioning_category_dict[surface.id] = "UNREGULATED"```  
+
+**Returns** ```return surface_conditioning_category_dict```  
 
 ## get_opaque_surface_type
-Description:  This function would cycle through each surface and determine whether it is a wall, ceiling or floor.  
+Description: This function would determine whether it is a wall, ceiling or floor.  
+
+Inputs:
+  - **Surface**: The surface that needs to determine surface type.  
+
+Returns:
+- **opaque_surface_type**: The Opaque Surface Type [roof, heated slab-on-grade, unheated slab-on-grade, floor, above-grade wall, below-grade wall].  
+
+
+Logic:  
+
+- If surface tilt is more than or equal to 0 degree and less than 60 degrees, surface is classified as roof: ```if 0 <= surface.tilt < 60: surface_type = "ROOF"```  
+
+- Else if surface tile is more than 120 degrees and less than or equal to 180 degrees, surface is classified as floor: ```else if 120 < surface.tilt <= 180:```  
+
+  - Determine if surface is heated slab-on-grade: ```if ( surface.construction.has_radiant_heating ) AND ( surface.adjacent_to == "GROUND" ): surface_type = "HEATED SLAB-ON-GRADE"```  
+
+  - Else determine if surface is unheated slab-on-grade: ```else if surface.adjacent_to == "GROUND": surface_type = "UNHEATED SLAB-ON-GRADE"```  
+
+  - Else, surface is floor: ```else: surface_type = "FLOOR"```  
+
+- Else, surface is classified as wall: ```else:```  
+
+  - Determine if surface is below grade wall: ```if surface.adjacent_to == "GROUND": surface_type = "BELOW-GRADE WALL"```  
+
+  - Else determine if surface is above grade wall: ```else: surface_type = "ABOVE-GRADE WALL"```  
+
+**Returns** ```return surface_type```  
+
+## get_wwr
+
+Description: This function would determine window wall ratio for a building segment.  
+
+Inputs:
+  - **RMR**: The RMR that needs to determine window wall ratio.  
+
+Returns:
+- **window_wall_ratio**: The window wall ratio of the building segment.  
+
+Function Call:
+
+- get_zone_conditioning_category()
+- get_surface_conditioning_category()
+- get_opaque_surface_type()
+
```

Comment:
get_zone_conditioning_category() is referenced on Line 250.

Reply:
I might well be missing something, but I am not sure we need it there either because it's called from within get_surface_conditioning_category. So in WWR function we need to (1) call get_surface_conditioning_category to find all semi-heated and exterior surfaces, and then (2) call get_opaque_surface_type to identify all such surfaces that are walls.